doc/neps/nep-0018-array-function-protocol.rst

```diff
@@ -552,6 +632,43 @@ high level NumPy API from adopting this proposal.
 NOTE: If you are reading this NEP in its draft state and disagree,
 please speak up on the mailing list!
 
+A magic decorator that inspects type annotations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A more succinct alternative would be to write these overloads with a decorator
+that builds overloaded functions automatically. Hypothetically, this might even
+directly parse Python 3 type annotations, e.g., perhaps
+
+.. code:: python
+
+    @overload_for_array_function
+    def broadcast_to(array: ArrayLike
+                     shape: Tuple[int, ...],
+                     subok: bool = False):
+        ...  # continue with the definition of broadcast_to
+
+The decorator ``overload_for_array_function`` would be written in terms
+of ``try_array_function_override``, but would also need some level of magic
+for (1) access to the wrapper function (``np.broadcast_to``) for passing into
+``__array_function__`` implementations and (2) dynamic code generation
+resembling the `decorator library <https://github.com/micheles/decorator>`_
+to automatically write an overloaded function like the manually written
+implemenations above with the exact same signature as the original.
+Unfortunately, using the ``inspect`` module instead of code generation would
```

Comment:
Presumably the `inspect` module would be needed for code generation anyway, so these are not mutually exclusive. Perhap "Using the inspect module at runtime instead of at code-generation time"?

Reply:
Thanks. I haven't gotten to revising this section yet (I just moved it unchanged down to "Alternatives")